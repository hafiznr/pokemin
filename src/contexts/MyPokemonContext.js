import React, { createContext, useState } from 'react';

export const MyPokemonContext = createContext();

export const MyPokemonContextProvider = ({ children, initialData }) => {
  const [myPokemonData, setMyPokemonData] = useState(JSON.parse(initialData) || []);

  const addPokemon = (data) => {
    const newPokemonData = [...myPokemonData, data];

    setMyPokemonData(newPokemonData);

    localStorage.setItem('mypokemon', JSON.stringify(newPokemonData));
  }

  const removePokemon = (nickname) => {
    const newPokemonData = myPokemonData
      .filter(pokemon => pokemon.nickname !== nickname);

    setMyPokemonData(newPokemonData);

    localStorage.setItem('mypokemon', JSON.stringify(newPokemonData));
  }

  return (
    <MyPokemonContext.Provider 
      value={{ myPokemonData, addPokemon, removePokemon }}
    >
      {children}
    </MyPokemonContext.Provider>
  )
}
