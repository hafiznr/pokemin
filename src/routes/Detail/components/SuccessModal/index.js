/** @jsxRuntime classic */
/** @jsx jsx */
import { useState } from "react"
import { 
  Button, 
  Modal, 
  ModalContent, 
  ModalHeader, 
  ModalBody, 
  FormControl, 
  FormLabel, 
  Input, 
  ModalOverlay
} from '@chakra-ui/react';
import { jsx, css } from '@emotion/react';

const SuccessModal = ({
  isOpen,
  pokemon = '',
  image,
  onClose,
  onSubmit
}) => {
  const [nickname, setNickname] = useState('');
  const [isError, setIsError] = useState(false);

  const handleChange = (e) => {
    setNickname(e.target.value);

    if (isError){
      setIsError(false);
    }
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    if (!nickname) {
      return setIsError(true);
    }

    setNickname('');
    setIsError(false);
    onSubmit(nickname);
  }

  const center = css({
    textAlign: 'center'
  });

  const imgStyle = css({
    display: 'inline'
  })

  return (
    <Modal 
      isOpen={isOpen} 
      onClose={onClose} 
      closeOnOverlayClick={false} 
      size="xs"
      isCentered
    >
      <ModalOverlay />
      
      <ModalContent>
        <ModalHeader>
          <div css={center}>Gotcha! {pokemon} was caught!</div>
        </ModalHeader>
        <ModalBody>
          <div css={center}>
            <img css={imgStyle} src={image} alt={pokemon} loading="lazy"/>
            
            <form onSubmit={handleSubmit}>
              <FormControl>
                <FormLabel>
                  Please give a nickname to the captured {pokemon}!
                </FormLabel>
                <Input
                  type="text"
                  name="nickname"
                  value={nickname}
                  isInvalid={isError}
                  focusBorderColor={isError ? 'red.500' : 'blue.500'}
                  onChange={handleChange}  
                />
              </FormControl>
              <Button 
                type="submit" 
                margin="8px 0" 
                width="100%" 
                size="lg"
                colorScheme="twitter"
              >
                Submit
              </Button>
            </form>
          </div>
        </ModalBody>
      </ModalContent>
    </Modal>
  )
}

export default SuccessModal;