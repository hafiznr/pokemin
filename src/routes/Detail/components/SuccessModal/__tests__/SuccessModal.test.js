import { render } from '@testing-library/react';
import SuccessModal from '../';
import { MyPokemonContextProvider } from '../../../../../contexts/MyPokemonContext';

test('renders SuccessModal', () => {
  const { container } = render(
    <MyPokemonContextProvider initialData={null}>
      <SuccessModal/>
    </MyPokemonContextProvider>
  );

  expect(container).toMatchSnapshot();
});