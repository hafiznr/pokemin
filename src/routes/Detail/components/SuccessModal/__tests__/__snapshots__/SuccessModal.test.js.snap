// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SuccessModal 1`] = `<div />`;
